File: app/src/main/java/org/droidupnp/legacy/upnp/CallableRendererFilter.java
package org.droidupnp.legacy.upnp;

import com.m3sv.plainupnp.data.UpnpDevice;

public class CallableRendererFilter implements ICallableFilter {

    private UpnpDevice device;

    public void setDevice(UpnpDevice device) {
        this.device = device;
    }

    @Override
    public Boolean call() {
        return device.asService("RenderingControl");
    }
}

File: app/src/main/java/org/droidupnp/legacy/upnp/CallableRendererFilter.kt
package org.droidupnp.legacy.upnp

import com.m3sv.plainupnp.data.upnp.UpnpDevice

class CallableRendererFilter : CallableFilter {

    override var device: UpnpDevice? = null

    override fun call(): Boolean = device?.asService("RenderingControl") ?: false
}
